src/graphics/entity.rs

```diff
@@ -0,0 +1,151 @@
+use glium::implement_vertex;
+
+use crate::vector::Vector3;
+
+pub const ENTITY_VERTEX_SHADER: &str = r#"
+        #version 150
+
+        in vec3 position;
+        in mat4 world_matrix;
+
+        // The vertex shader has some passthrough for the fragment shader...
+
+        // Which face of the cube is being passed ?
+        in int face;
+        flat out int face_s;
+
+        // Index of the block to be used
+        in int body_part_id;
+        flat out int body_part_id_s;
+
+        // Where is the vertex located on the face ?
+        in vec2 tex_coords;
+        out vec2 v_tex_coords;
+
+        uniform mat4 perspective;
+        uniform mat4 view;
+
+        void main() {
+            gl_Position = perspective * view * world_matrix * vec4(position, 1.0);
+            v_tex_coords = tex_coords;
+            face_s = face;
+            body_part_id_s = body_part_id;
+        }
+    "#;
+
+// Fragment shader
+pub const ENTITY_FRAGMENT_SHADER: &str = r#"
+        #version 140
+
+        // passed-through the vertex shader
+        flat in int face_s;
+        flat in int body_part_id_s;
+        in vec2 v_tex_coords;
+
+        out vec4 color ;
+
+        uniform sampler2DArray textures;
+        
+        void main() {
+            // Each block has 6 types of faces
+            int idx = body_part_id_s * 6;
+            color = texture(textures, vec3(v_tex_coords, idx + face_s));
+         }
+    "#;
+
+
+/// A vertex of a cube
+/// The position is expressed into the OpenGL reference frame
+#[derive(Copy, Clone)]
+pub struct EntityVertex {
+    position: [f32; 3],
+    tex_coords: [f32; 2],
+    face: u8,
+}
+
+implement_vertex!(EntityVertex, position, tex_coords, face);
+
+/// This is different from CubeVertex, the face numbering have been changed
+pub const ENTITY_VERTICES: [EntityVertex; 36] = [
```

Comment:
Ok this one must be removed, it's basically a duplicate of the other one. You can change the "face" of the other ones if this format suits you better.

Reply:
Yes i will copy-paste this one in the old one but then don't change the vertices' position as you wrote in a TODO !